pinnable.go

```diff
@@ -0,0 +1,38 @@
+package ebpf
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cilium/ebpf/internal"
+)
+
+func pin(currentPath, newPath string, fd *internal.FD) error {
+	if newPath == "" {
+		return fmt.Errorf("pinned path cannot be empty")
+	}
+	if currentPath != "" {
+		if currentPath == newPath {
+			return nil
+		}
+		if err := os.Rename(currentPath, newPath); err != nil {
+			if !os.IsNotExist(err) {
+				return fmt.Errorf("unable to pin the object at new path %v: %w", newPath, err)
+			}
+		} else {
+			return nil
+		}
+	}
+	return internal.BPFObjPin(newPath, fd)
```

Comment:
Executing the pin operation as a fall-through makes it difficult to figure out at first sight in which cases the pin syscall is executed. I would put this right after `if newpath == "" { ... }` at the top:
```go
if currentpath == "" {
  return internal.BPFObjPin(newPath, fd)
}
```